presidio-analyzer/analyzer/analyzer_engine.py

```diff
@@ -117,6 +118,24 @@ def __convert_fields_to_entities(self, fields):
             entities.append(field.name)
         return entities
 
+    def get_fields_from_template(self, analyzeTemplate):
+        """
+        Returns a list of fields. If the "allFields" flag if True, all fields will be returned. Otherwise,
+        the template fields will be returned.
+
+        :param analyzeTemplate: The template to process.
+        :return: a list of fields
+        """
+        if analyzeTemplate.allFields:
+            fields = []
+            for item in common_pb2.FieldTypesEnum.keys():
```

Comment:
This is not going to work for custom recognizers. the allFields param should skip the filtering of recognizers by field